### PR TITLE
Security fixes

### PR DESCRIPTION
**Pull request by datnguyen-sc**
## Description

Fix issues highlighted by HM-Minimum coding standards.

### CODE REVIEW

File: functions/backendpreparer.php [hunk 1]
```diff
@@ -129,19 +129,19 @@ function handleFeaturedImageBox() {
 		 */
 		var baseElem = $('#postimagediv');
 		if(!baseElem.length) { return; }//this is not WordPress < 5
-		
+
 		var featuredImageLinkButton = '';
 		featuredImageLinkButton+= '<p class="cropFeaturedImageWrap hidden">';
-		featuredImageLinkButton+= '<a class="button cropThumbnailsLink" href="#" data-cropthumbnail=\'{"image_id":'+ parseInt(wp.media.featuredImage.get()) +',"viewmode":"single","posttype":"<?php echo get_post_type(); ?>"}\' title="<?php esc_attr_e('Crop Featured Image','crop-thumbnails') ?>">';
+		featuredImageLinkButton+= '<a class="button cropThumbnailsLink" href="#" data-cropthumbnail=\'{"image_id":'+ parseInt(wp.media.featuredImage.get()) +',"viewmode":"single","posttype":"<?php echo esc_attr( get_post_type() ); ?>"}\' title="<?php esc_attr_e('Crop Featured Image','crop-thumbnails') ?>">';
```
Comment (Nikschavan) [inline functions/backendpreparer.php]: I believe this should be `esc_js()`

```suggestion
		featuredImageLinkButton+= '<a class="button cropThumbnailsLink" href="#" data-cropthumbnail=\'{"image_id":'+ parseInt(wp.media.featuredImage.get()) +',"viewmode":"single","posttype":"<?php echo esc_js( get_post_type() ); ?>"}\' title="<?php esc_attr_e('Crop Featured Image','crop-thumbnails') ?>">';
```

File: functions/backendpreparer.php [hunk 2]
```diff
@@ -152,34 +152,34 @@ function updateCropFeaturedImageButton(currentId) {
...
 		if(typeof wp.element === 'undefined') { return };//this is not WordPress 5.x
 
 		var el = wp.element.createElement;
-		function wrapPostFeaturedImage( OriginalComponent ) { 
+		function wrapPostFeaturedImage( OriginalComponent ) {
 			return function( props ) {
 				setTimeout(function() {
 					var baseElem = $('.edit-post-sidebar');
-					var cropButton = $('<button class="button cropThumbnailsLink" style="margin-top:1em" data-cropthumbnail=\'{"image_id":'+ parseInt(props.featuredImageId) +',"viewmode":"single","posttype":"<?php echo get_post_type(); ?>"}\' title="<?php esc_attr_e('Crop Featured Image','crop-thumbnails') ?>"><span class="wp-media-buttons-icon"></span> <?php esc_html_e('Crop Featured Image','crop-thumbnails'); ?></button>');
+					var cropButton = $('<button class="button cropThumbnailsLink" style="margin-top:1em" data-cropthumbnail=\'{"image_id":'+ parseInt(props.featuredImageId) +',"viewmode":"single","posttype":"<?php echo esc_attr( get_post_type() ); ?>"}\' title="<?php esc_attr_e('Crop Featured Image','crop-thumbnails') ?>"><span class="wp-media-buttons-icon"></span> <?php esc_html_e('Crop Featured Image','crop-thumbnails'); ?></button>');
```
Comment (Nikschavan) [inline functions/backendpreparer.php]: Same as above, the change should be `esc_js` I believe.

File: functions/settingsscreen.php [hunk 3]
```diff
@@ -203,31 +232,31 @@ public function validateSettings($input) {
...
 					$('#cpt_quicktest').remove();
-					
+
 					$.ajax({
 						url: ajaxurl,
 						type: 'POST',
 						data: {
 							action: 'ctppluginquicktest',
-							security: '<?php echo wp_create_nonce( "cpt_quicktest-ajax-nonce" );//only for quicktest ?>'
+							security: '<?php echo esc_attr( wp_create_nonce( "cpt_quicktest-ajax-nonce" ) );//only for quicktest ?>'
```
Comment (Nikschavan) [inline functions/settingsscreen.php]:
```suggestion
							security: '<?php echo esc_js( wp_create_nonce( "cpt_quicktest-ajax-nonce" ) );//only for quicktest ?>'
```

**Review by Nikschavan — CHANGES_REQUESTED** [review]
I think we should revert the indentation/white-space changes (either in this PR or I can do this in a separate one later) so that it is easier to get this merged into the upstream repo.

The plan is to get the security fixes merged to upstream in the near term so that we don't end up maintaining the fork for a few fixes.

**Comment by datnguyen-sc** [comment]
@Nikschavan: Changed `esc_attr` to `esc_js`.  For the spacing, I think it should be alright as all the changes to spacing are to remove unnecessary spaces. I'm not changing spacing character, e.g. space to tab, etc.

**Review by Nikschavan — APPROVED** [review]

**Pull request merged**